packages/form-js-viewer/src/render/components/form-fields/parts/SimpleSelect.js

```diff
@@ -0,0 +1,84 @@
+import { useCallback, useContext, useMemo, useRef, useState } from 'preact/hooks';
+import useValuesAsync, { LOAD_STATES } from '../../../hooks/useValuesAsync';
+
+import { FormContext } from '../../../context';
+
+import XMarkIcon from '../icons/XMark.svg';
+import AngelDownIcon from '../icons/AngelDown.svg';
+import AngelUpIcon from '../icons/AngelUp.svg';
+
+import {
+  prefixId,
+} from '../../Util';
+
+import classNames from 'classnames';
+
+import DropdownList from './DropdownList';
+import { findIndex } from 'min-dash';
+
+export default function SimpleSelect(props) {
+  const {
+    id,
+    disabled,
+    errors,
+    field,
+    value
+  } = props;
+
+  const { formId } = useContext(FormContext);
+  const [ isDropdownExpanded, setIsDropdownExpanded ] = useState(false);
+  const selectRef = useRef();
+
+  const {
+    state: loadState,
+    values: options
+  } = useValuesAsync(field);
+
+  // We cache a map of option values to their index so that we don't need to search the whole options array every time to correlate the label
+  const valueToOptionMap = useMemo(() => Object.assign({}, ...options.map((o, x) => ({ [o.value]: options[x] }))), [ options ]);
+
+  const valueLabel = useMemo(() => value && valueToOptionMap[ value ]?.label || '', [ value, valueToOptionMap ]);
```

Comment:
I believe we shouldn't use nullish coalescing, right @pinussilvestrus ?

Reply:
I'd consider this a code style preference and if it's not in the linting rules, it's should be allowed IMO. Now we could make a case that this should be linted, but personally, I find it much more readable than the alternative.

Reply:
It's not just code style preference because we support only `ES2018` and nullish coallescing is `ES2020` and we don't have any pollyfills in our build pipeline

Reply:
@vsgoulart Should I then get rid of all optional chaining usages? I feel like it should have popped up as in issue in the past 2 months ([commit](https://github.com/bpmn-io/form-js/commit/67dc09f069bf26632ba6c01f3be566ac412163b8)).

Reply:
We could an Eslint rule for that, so we don't have to worry about it on reviews

Reply:
Last time we checked, nullish coalescing is indeed something we should avoid, as it is ES2020, as @vsgoulart mentioned (cf. https://github.com/bpmn-io/form-js/pull/448). Optional chaining we use already in the code base, e.g. [here](https://github.com/bpmn-io/form-js/blob/develop/packages/form-js-editor/src/features/properties-panel/groups/ValidationGroup.js#L38). Still, I wonder why this doesn't blow up, isn't it also ES2020? Most likely, we transpile it via rollup, so we are safe. That's not the case for nullish coalescing.

Personally, I'd like to avoid these "newer" JS features, also for consistency reasons.

Reply:
@pinussilvestrus We don't transpile it actually, you can find the optional chaining usages down to the desktop modeler code.